pkg/controller/constraint/constraint_controller.go

```diff
@@ -303,8 +300,27 @@ func (r *ReconcileConstraint) Reconcile(ctx context.Context, request reconcile.R
 		status.Status.Errors = nil
 
 		if c, err := r.cfClient.GetConstraint(instance); err != nil || !reflect.DeepEqual(instance, c) {
+			generateVAPB, VAPEnforcementActions, err := shouldGenerateVAPB(*DefaultGenerateVAPB, enforcementAction, instance)
+			if err != nil {
+				status.Status.Errors = append(status.Status.Errors, constraintstatusv1beta1.Error{Message: err.Error()})
+				log.Error(err, "could not get enforcement actions for VAP")
+				if err2 := r.writer.Update(ctx, status); err2 != nil {
+					log.Error(err2, "could not report error for getting enforcement actions for VAP")
+				}
+				return reconcile.Result{}, err
+			}
+			isVAPBGenerationEnabled := generateVAPB && IsVapAPIEnabled() && HasVAPCel(ct)
+			if generateVAPB != isVAPBGenerationEnabled {
```

Comment:
We should probably just calculate these errors directly, rather than packing multiple values into a bool and then unpacking them again.